app/graphql/resolvers/submission_resolver.rb

```diff
@@ -1,19 +1,20 @@
 # frozen_string_literal: true
 
 class SubmissionResolver < BaseResolver
-  def valid?
-    unless admin?
-      bad_argument_error =
-        GraphQL::ExecutionError.new("Can't access submission")
-      @error = bad_argument_error
+  def run
+    submission = Submission.find_by(id: @arguments[:id])
+
+    unless submission
+      raise GraphQL::ExecutionError, 'Submission from ID Not Found'
     end
 
-    admin? || partner?
-  end
+    unless (
+             matching_user(submission, @arguments&.[](:session_id)) &&
+               submission.draft?
```

Comment:
**suggestion:** if the check for `submission.draft?` comes before the call to `matching_user`, this resolver will be a little faster.